### PR TITLE
Fix "int_sequence not declared" error when build Neon with C++14

### PR DESCRIPTION
**Pull request by cyb70289**

### CODE REVIEW

File: include/xsimd/types/xsimd_utils.hpp [hunk 1]
```diff
@@ -240,6 +240,7 @@ namespace xsimd
 
             template <typename... Ts>
             using index_sequence_for = make_index_sequence<sizeof...(Ts)>;
+        #endif
```
Comment (cyb70289) [inline include/xsimd/types/xsimd_utils.hpp:243]: Symbols below should be available no matter c++ version.
Original code only defines those symbols under c++11, and causes error when build xsimd neon if -std=c++14.
`xsimd::detail::int_sequence’ has not been declared`

Comment (serge-sans-paille) [reply include/xsimd/types/xsimd_utils.hpp:243]: The `int_sequence` code looks super redundant with the `index_sequence` code. I'll propose a patch that make the whole stuff more generic and take your remark into account.

**Comment by serge-sans-paille** [comment]
Can you confirm #580 fixes the issue?

**Comment by cyb70289** [comment]
> Can you confirm #580 fixes the issue?

Yes, #580 fixes this issue. Thanks.

**Pull request closed without merging**